jina/drivers/evaluate.py

```diff
@@ -90,28 +91,35 @@ def extract(self, doc: 'Document') -> Any:
         return dunder_get(doc, self.field)
 
 
-class RankEvaluateDriver(FieldEvaluateDriver):
+class RankEvaluateDriver(BaseEvaluateDriver):
     """Drivers used to pass `matches` from documents and groundtruths to an executor and add the evaluation value
 
         - Example fields:
-        ['tags__id', 'id', 'score__value]
+            ['tags__id', 'score__value]
+
+    :param fields: the fields names to be extracted from the Protobuf.
+            The differences with `:class:FieldEvaluateDriver` are:
+                - More than one field is allowed. For instance, for NDCGComputation you may need to have both `ID` and `Relevance` information.
+                - The fields are extracted from the `matches` of the `Documents` and the `Groundtruth` so it returns a sequence of values.
+    :param *args:
+    :param **kwargs:
     """
 
     def __init__(self,
-                 field: str = 'tags__id',
+                 fields: Tuple[str] = ('tags__id',),
                  *args,
                  **kwargs):
-        """
-        :param field: the field name to be extracted from the Protobuf
-        :param *args:
-        :param **kwargs:
-        """
-        super().__init__(field, *args, **kwargs)
+
+        super().__init__(*args, **kwargs)
+        self.fields = fields
 
     def extract(self, doc: 'Document'):
-        r = [dunder_get(x, self.field) for x in doc.matches]
-        # flatten nested list but useless depth, e.g. [[1,2,3,4]]
-        return list(numpy.array(r).flat)
+        if len(self.fields) > 1:
+            ret = [tuple(dunder_get(x, field) for field in self.fields) for x in doc.matches]
+        else:
+            ret = [dunder_get(x, self.fields[0]) for x in doc.matches]
```

Comment:
`self.fields=[]` will lead to an exception

Reply:
Having the returning type depending on a class member variable doesn't look good. Can we always return a list of tuples?

Reply:
Remember to keep this part backward compatible.

Reply:
It is an option, but it will make it quite dirty for the executors? I also do not like it so much

Reply:
Can u check the changes? I did it backward compatible, but I do not check that there is not None in the Tuple (since it is not an Optional as per signature I am not sure it is so useful to check.

The same way we do not check if `field` is not None